### ISSUE TITLE
CloudTrail can't validate ARN of CloudWatch Logs LogGroup

### ISSUE THREAD

**Issue by marcin-contino** [terraform-providers/terraform-provider-aws#14557]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.12.29
AWS provider version: 3.0.0
OSX: 10.15.6

### Affected Resource(s)

CloudTrail, CloudWatch Logs

### Terraform Configuration Files

The same code does work OK with AWS provider version 2.70.0

### Expected Behavior

CloudTrail to validate the ARN of CloudWatch Logs LogGroup

### Actual Behavior

```
Error: Error creating CloudTrail: InvalidCloudWatchLogsLogGroupArnException: Check the log group ARN: CloudTrail can't validate it.

  on ../modules/tf-module-cloudtrail/40-cloudtrail.tf line 1, in resource "aws_cloudtrail" "cloudtrail-logging":
   1: resource "aws_cloudtrail" "cloudtrail-logging" {
```

### Steps to Reproduce

1. `terraform apply`

**Comment by bflad** [comment]
Hi @marcin-contino 👋  Thank you for filing this and sorry you ran into trouble here. Please see the [Version 3 Upgrade Guide section on the `aws_cloudwatch_log_group` resource]() for the expected change in the new major version. The CloudWatch Log Group ARN previously contained a `:*` suffix which was incompatible with many other AWS services, however there are some like CloudTrail and IAM that sometimes expect it. To fix your configuration on 3.0.0:

```terraform
resource "aws_cloudtrail" "example" {
  # ... other configuration ...

  cloud_watch_logs_group_arn = "${aws_cloudwatch_log_group.example.arn}:*" # CloudTrail requires the Log Stream wildcard
}
```

I am going to leave this issue open for anyone to update the `aws_cloudtrail` resource documentation (`website/docs/r/cloudtrail.html.markdown`) to include this additional information in the `cloud_watch_logs_group_arn` argument documentation and potentially create another example configuration.

**Comment by sebolabs** [comment]
Still got it with 3.5.0

```
Initializing provider plugins...
- Finding latest version of hashicorp/aws...
- Using hashicorp/aws v3.5.0 from the shared cache directory

[...]

* hashicorp/aws: version = "~> 3.5.0"

[...]

Error creating CloudTrail: InvalidCloudWatchLogsLogGroupArnException: Check the log group ARN: CloudTrail can't validate it.
```

had to add `:*` to make it work

**Comment by bflad** [comment]
Hi @sebolabs 👋  To clarify the closure above -- it is for the documentation update that was released in 3.3.0. The `:*` suffix is required after 3.0.0: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail#cloud_watch_logs_group_arn

**Comment by sebolabs** [comment]
Oh, my bad 👍🏻 Thanks
